pkg/build/builder/daemonless.go

```diff
@@ -45,6 +56,64 @@ func dropCapabilities() []string {
 	return dropCapabilities
 }
 
+// parsePullCredentials parses credentials from provided file.
+func parsePullCredentials(credsPath string) (credentialprovider.DockerConfig, error) {
+	var creds credentialprovider.DockerConfig
+	var err error
+
+	if filepath.Base(credsPath) == dockercfg.DockerConfigKey {
+		if creds, err = credentialprovider.ReadDockercfgFile(
+			[]string{filepath.Dir(credsPath)},
+		); err != nil {
+			return nil, err
+		}
+	} else {
+		if creds, err = credentialprovider.ReadSpecificDockerConfigJsonFile(
+			credsPath,
+		); err != nil {
+			return nil, err
+		}
+	}
+
+	if creds == nil {
+		creds = make(map[string]credentialprovider.DockerConfigEntry)
+	}
+
+	return creds, nil
+}
+
+// mergeNodeCredentials merges node credentials with credentials file provided.
+//
+// Node pull credentials are merged with credentials present on file path
+// provided as parameter, output is written to dstFile.
+func mergeNodeCredentials(credsPath string, dstFile io.Writer) error {
```

Comment:
This function should probably just actually merge the credentials and then return the merged map, and have the calling function write them to the file?

Reply:
OK, done.